lib/cylc/gui/tailer.py

```diff
@@ -74,36 +67,13 @@ def clear(self):
         self.logbuffer.delete(pos_start, pos_end)
 
     def run(self):
-        """Invoke the tailer."""
-        command = []
-        cmd_tmpl = None
-        if self.cmd_tmpl:
-            cmd_tmpl = self.cmd_tmpl
-        if ":" in self.filename:  # remote
-            user_at_host, filename = self.filename.split(':')
-            if "@" in user_at_host:
-                owner, host = user_at_host.split("@", 1)
-            else:
-                owner, host = (None, user_at_host)
-            ssh = str(glbl_cfg().get_host_item("ssh command", host, owner))
-            command = shlex.split(ssh) + ["-n", user_at_host]
-            if not cmd_tmpl:
-                cmd_tmpl = str(glbl_cfg().get_host_item(
-                    "remote tail command template", host, owner))
-            command.append(cmd_tmpl % {"filename": filename})
-        else:
-            filename = self.filename
-            if not cmd_tmpl:
-                cmd_tmpl = str(glbl_cfg().get_host_item(
-                    "local tail command template"))
-            command += shlex.split(cmd_tmpl % {"filename": filename})
-
+        """Invoke the command."""
+        command = shlex.split(self.cmd)
         try:
             self.proc = Popen(
                 command, stdin=open(os.devnull), stdout=PIPE, stderr=STDOUT,
-                preexec_fn=os.setpgrp)
+                preexec_fn=os.setsid)
```

Comment:
Could you quickly explain why we need the process to be a session leader instead of just a process group leader?

Reply:
I think I found by experimentation this was needed to get all processes to die on exit - but I could be wrong, and I don't remember if, sorry.  I'll take another look ...

Reply:
Looks like I was wrong about this - revered to `setpgrp`.